File: 卡码/代码随想录/1/06-二叉树/Go/02-二叉树属性/LeetCode_0111.go
//111. 二叉树的最小深度
package main

import (
	. "binaryTree/types"
)

func minDepth(root *TreeNode) int {
	if root==nil {
		return 0
	}
	if root.Left==nil&&root.Right!=nil {
		return 1+min(minDepth(root.Right))
	}
	if root.Left!=nil&&root.Right==nil {
		return 1+min(minDepth(root.Left))
	}
	return 1+min(minDepth(root.Left),minDepth(root.Right))
}

File: 卡码/代码随想录/1/06-二叉树/Python/02-二叉树属性/LeetCode_0111.py
#111. 二叉树的最小深度
from typing import *
from ..TreeNode import *

class Solution:
  def minDepth(self, root: Optional[TreeNode]) -> int:
    if not root:
      return 0
    if root.left and not root.right:
      return 1+self.minDepth(root.left)
    if not root.left and root.right:
      return 1+self.minDepth(root.right)
    return 1+min(self.minDepth(root.left),self.minDepth(root.right))
